Update glossary doc with terminology (#1100)
kubernetes-sigs/cluster-api: docs/book/GLOSSARY.md

* __Core Add-ons__: Addons that are required to deploy a Kubernetes-conformant cluster: DNS, kube-proxy, CNI.
* __Additional Add-ons__: Addons that are not required for a Kubernetes-conformant cluster (e.g. metrics/Heapster, Dashboard).

# B

## Bootstrap

The process of turning a server into a Kubernetes node. This may involve assembling data to provide when creating the server that backs the Machine, as well as runtime configuration of the software running on that server.

## Bootstrap cluster

> A temporary cluster that is used to provision a Target cluster.

Reviewer: Since @dlipovetsky brought up the Management/Workload cluster nomenclature, and to avoid causing confusion with those terms, maybe this should call out `Target Management cluster` rather than just `Target cluster`

> A temporary cluster to provision the Target cluster, specified in the related resource files.

Reviewer: ```suggestion
A temporary cluster that is used to provision a Target cluster.
```

# P

## Pivot

> Pivot deploys the provided provider components to a Target cluster and then migrates all Cluster API resources from the Bootstrap cluster to the Target cluster.

Reviewer: I would modify this to something along the lines of: Pivot is a process for moving the provider components and declared cluster-api resources from a source cluster to a target cluster.

The pivot process is also used for deleting a management cluster and could also be used during an upgrade of the management cluster.

# P

## Pivot

> Pivot is a process for moving the provider components and declared cluster-api resources from a source cluster to a target cluster.

Reviewer: ```suggestion
Pivot is a process for moving the provider components and declared cluster-api resources from a Source cluster to a Target cluster.
```

> Pivot is a process for moving the provider components and declared cluster-api resources from a Source cluster to a Target cluster.

Reviewer: Let's update this to `Source Management cluster` and `Target Management cluster`

The infrastructure that backs a [Machine Resource](#user-content-machine), typically either a cloud instance, virtual machine, or physical host.

# T

## Target cluster

> The declared cluster we intend to create and manage using Bootstrap cluster.

Reviewer: If pivot is redefined to remove the hard reference to the bootstrap cluster, then this definition will need to be modified as well.

Author: Hi @detiber ,thanks for replying.
I agree with you on the pivot definition changes but confused regarding this change. I may be wrong but is it not that the term "target cluster" is used only in the case where we are using a bootstrap cluster. Because there need to be a difference between target and a workload cluster.

Reviewer: If describing the `clusterctl create` workflow it is true that you have bootstrap cluster and a target cluster, but in the context of just pivot the source and target clusters can be any pre-existing Kubernetes cluster. It gets even more complicated if discussing the `clusterctl delete` workflow, since the source cluster for pivot is the remote cluster and the target cluster is the "bootstrap" cluster.

Author: Ohh okay. I will update it. Thanks @detiber

The infrastructure that backs a [Machine Resource](#user-content-machine), typically either a cloud instance, virtual machine, or physical host.

# T

## Target cluster

> The declared cluster we intend to create and manage using source cluster.

Reviewer: ```suggestion
The declared cluster we intend to create and manage using cluster-api when running `clusterctl create cluster`.

When running `clusterctl alpha phases pivot` this refers to the cluster that will be the new management cluster.
```

Reviewer 2: Thanks for the work on the glossary! However, I thought that we have been using the terms "Management" and "Workload" to refer to clusters? Are these terms not expressive enough?

Reviewer: They are not expressive enough to discuss the process of pivoting a management cluster.

Reviewer 2: Good point

The infrastructure that backs a [Machine Resource](#user-content-machine), typically either a cloud instance, virtual machine, or physical host.

# T

> ## Target cluster

Reviewer: Let's update to `Target Management cluster`
